### PR TITLE
Fix text when switching out after failing a switch move

### PR DESCRIPTION
**Pull request by Karthik99999**

### CODE REVIEW

File: src/battle.ts [hunk 1]
```diff
@@ -867,15 +866,16 @@ export class Side {
...
+		const effect = Dex.getEffect(kwArgs.from);
 		if (pokemon.lastMove !== 'batonpass' && pokemon.lastMove !== 'zbatonpass') {
 			pokemon.clearVolatile();
 		} else {
 			pokemon.removeVolatile('transform' as ID);
 			pokemon.removeVolatile('formechange' as ID);
 		}
-		if (pokemon.lastMove === 'uturn' || pokemon.lastMove === 'voltswitch') {
-			this.battle.log(['switchout', pokemon.ident], {from: pokemon.lastMove});
+		if (['uturn', 'voltswitch', 'flipturn'].includes(effect.id)) {
```
Comment (DaWoblefet) [inline src/battle.ts:877]: Is there a way we can generalize this better? This is missing Parting Shot too, for example.

Comment (Karthik99999) [reply src/battle.ts:877]: how about this?
```js
if (pokemon.lastMove !== 'batonpass' && pokemon.lastMove !== 'zbatonpass') {
	this.battle.log(['switchout', pokemon.ident], {from: kwArgs.from});
}
```
`BattleTextParser` should automatically deal with the `from` effect not having its own switch out text (or 'from` not existing

Comment (DaWoblefet) [reply src/battle.ts]: Your solution here looks good.

**Review by Marty-D — APPROVED** [review]
Thanks! 👍

**Pull request merged**